File: public/code/G9DbjnwIgay7UeW9Sti3/palindrome.py

def is_palindrome(s):
    left, right = 0, len(s) - 1
    while left < right:
        if s[left] != s[right]:
            return False
        left += 1
        right -= 1
    return True

s = input("Enter a string: ")
if is_palindrome(s):
    print("Is Palindrome")
else:
    print("Not a Palindrome")

File: public/code/G9DbjnwIgay7UeW9Sti3/palindrome.cpp
#include <iostream>
#include <string>
using namespace std;

bool isPalindrome(string str) {
    int left = 0, right = str.length() - 1;
    while (left < right) {
        if (str[left] != str[right])
            return false;
        left++;
        right--;
    }
    return true;
}

int main() {
    string str;
    cout << "Enter a string: ";
    cin >> str;
    if (isPalindrome(str)) {
        cout << "Is Palindrome" << endl;
    } else {
        cout << "Not a Palindrome" << endl;
    }
    return 0;
}
